testing/test_has_context.py

```diff
@@ -0,0 +1,92 @@
+import os
```

Comment:
test_has_context.py could be renamed if the class is renamed as I suggest above.